sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/util/ModelHelper.java

```diff
@@ -196,4 +200,28 @@ public static ShareFileItemProperties transformFileProperty(FileProperty propert
         return new InternalShareFileItemProperties(property.getCreationTime(), property.getLastAccessTime(),
             property.getLastWriteTime(), property.getChangeTime(), property.getLastModified(), property.getEtag());
     }
+
+    public static Response<ShareFileProperties> buildShareFilePropertiesResponse(ShareFileDownloadAsyncResponse response) {
+        ShareFileDownloadHeaders headers = response.getDeserializedHeaders();
+        ShareFileProperties properties = new ShareFileProperties(
+            headers.getETag(),
+            headers.getLastModified(),
+            headers.getMetadata(),
+            headers.getContentType(),
+            headers.getContentLength(),
+            headers.getContentType(),
+            headers.getContentMd5(),
+            headers.getContentEncoding(),
+            headers.getCacheControl(),
+            headers.getContentDisposition(),
+            headers.getCopyCompletionTime(),
+            headers.getCopyStatusDescription(),
+            headers.getCopyId(),
+            headers.getCopyProgress(),
+            headers.getCopySource(),
+            headers.getCopyStatus(),
+            headers.isServerEncrypted(),
+            null);
```

Comment:
One minor worry is customers who were using `FileSmbProperties` from download before will no longer have access to them.